src/Exceptions/ApiException.php

```diff
@@ -85,4 +87,9 @@ private function getErrorLinkFromHttpBody(): ?string
 
         return null;
     }
+
+    public static function rethrowWithHint(\Exception $e, string $methodName)
+    {
+        return new \Exception(sprintf(self::HINT_MESSAGE, $methodName), 0, $e);
```

Comment:
why instead of throwing ApiException as you define the static constructor in it, you throw generic exception?

Reply:
Good question. I decided because it has the simplest constructor. All Meilisearch error classes require a `ResponseInterface $response` to be sent alongside it. To make that happen, I would have to deal with the `$e` and make some validations before rethrowing too much work for no apparent gain.